novawallet/Common/DataProvider/Subscription/Proxy/ProxyListLocalSubscriptionFactory.swift

```diff
@@ -2,24 +2,40 @@ import RobinHood
 
 protocol ProxyListLocalSubscriptionFactoryProtocol {
     func getProxyListProvider() throws -> StreamableProvider<ProxyAccountModel>
+    func getProxyListProvider(
+        for accountId: AccountId,
+        chainId: ChainModel.Id
+    ) throws -> AnyDataProvider<DecodedProxyDefinition>
 }
 
 final class ProxyListLocalSubscriptionFactory: BaseLocalSubscriptionFactory {
     static let shared = ProxyListLocalSubscriptionFactory(
```

Comment:
Let's not make it shared because it used only in staking and no sense to store otherwise. I would store it in the state as StakingLocalSubscriptionFactory

Reply:
It's also used in walletNotificationService